themes/ddbasic/sass/components/header.scss

```diff
@@ -3,702 +3,451 @@
 
 @import '../base.scss';
 
-// In order to make the theme backwards compatible we are styling the existing markup.
-// This makes the code in this file a bit messy.
 .site-header {
-  // ==========================================================================
-  // Topbar
-  // ==========================================================================
-  .topbar {
-    @include transition(transform $speed $ease);
-    @include transform(translateY(0%));
-    position: fixed;
-    z-index: $z-overlay + 1;
-    top: 0;
-    left: 0;
-    width: 100%;
-    background-color: $grey-light;
-    box-shadow: $box-shadow;
+  position: fixed;
+  z-index: $z-overlay + 1;
+  top: 0;
+  left: 0;
+  width: 100%;
+  .admin-menu & {
+    top: $admin-menu-height;
+  }
+  .site-header-inner {
+    @include wrapper;
 
     // Tablet
     @include media($tablet) {
-      box-shadow: none;
-    }
-    // Admin menu
-    .admin-menu & {
-      top: $admin-menu-height;
-    }
-    // Admin menu shortcuts or toolbar.
-    .admin-menu-with-shortcuts &,
-    .toolbar & {
-      top: $toolbar-height;
+      width: 100%;
     }
   }
-  .topbar-inner {
-    @include wrapper;
-    @include transition(height $speed $ease);
-    position: relative;
-    height: 64px;
+}
+
+// ==========================================================================
+// Topbar
+// ==========================================================================
+.site-header-topbar {
+  position: relative;
+  z-index: $z-overlay + 1;
+  background-color: $grey-light;
+  box-shadow: $box-shadow;
+
+  .pane-header-logo {
+    @include span-columns(4);
+    height: 44px;
+    margin-top: 10px;
 
     // Tablet
     @include media($tablet) {
+      position: relative;
+      width: 100%;
+      height: 64px;
+      margin-top: 0;
+      padding: 10px 5%;
+      box-shadow: $box-shadow;
+      background-color: $grey-light;
+      z-index: 5;
+    }
+  }
+
+  // Tablet
+  @include media($tablet) {
+    box-shadow: none;
+    .topbar-menu {
+      position: relative;
+      z-index: 3;
       width: 100%;
+      background-color: $color-primary;
+      padding: 10px 5%;
+      box-shadow: $box-shadow;
     }
+  }
+}
 
-    .logo {
-      @include span-columns(4);
-      height: 44px;
-      margin-top: 10px;
+// ==========================================================================
+// Search and user menu
+// ==========================================================================
+.site-header-search-and-user-menu {
+  position: relative;
+  z-index: $z-overlay;
 
-      // Tablet
-      @include media($tablet) {
-        position: relative;
-        width: 100%;
-        height: 64px;
-        margin-top: 0;
-        padding: 10px 5%;
-        box-shadow: $box-shadow;
-        background-color: $grey-light;
-        z-index: 5;
+  .pane-user-menu {
+    position: absolute;
+    top: 0;
+    z-index: 51;
+
+    // Tablet
+    @include media($tablet) {
```

Comment:
This use of comments makes little sense. The comment does not provide any value to the subsequent code. 

This pattern seems to go through the majority of the changes for media queries and other elements.